### PR TITLE
Dgds return value at point

### PR DESCRIPTION
**Pull request by ckrogers**
endpoint /get_feature_info returns value of imageId at bbox geojson

### CODE REVIEW

File: hydroengine_service/main.py [hunk 1]
```diff
@@ -1820,6 +1815,11 @@ def generate_image_info(im, params):
         n_colors = len(params.get('palette'))
         palette = params.get('palette')
         offsets = np.linspace(0, 100, num=n_colors)
+        if 'function' in params:
+            if params['function'] == 'log':
+                # offsets
+                offsets = np.logspace(0.0, 2.0, num=n_colors, base=10.0)
```
Comment (SiggyF) [inline hydroengine_service/main.py]: Please explain

File: hydroengine_service/main.py [hunk 2]
```diff
@@ -1834,6 +1834,68 @@ def generate_image_info(im, params):
...
+    if operation == "flowmap":
+        image.unitScale(data_params['min'][operation], data_params['max'][operation]).unmask(-9999)
+        data_mask = image.eq(-9999).select(data_params['bandNames'][band])
+        image = image.clamp(0, 1).addBands(data_mask).rename('flowmap')
+
+    return image
+
+@app.route('/get_feature_info', methods=['POST'])
+@flask_cors.cross_origin()
+def get_feature_info():
```
Comment (SiggyF) [inline hydroengine_service/main.py]: Did you add a test for this in postman?

File: hydroengine_service/main.py [hunk 3]
```diff
@@ -1834,6 +1834,68 @@ def generate_image_info(im, params):
...
+        if len(image_location_parameters) == 4:
+            _gee_folder_type, _project, data_collection, dataset = image_location_parameters
+
+        band_name = DATASETS_VIS[data_collection][dataset]['bandNames'][band]
+        image = image.select(band_name)
+
+    image = apply_image_operation(image, function)
+    image = image.rename('value')
+
+    value = image.sample(**{
```
Comment (SiggyF) [inline hydroengine_service/main.py]: If  you put python code in `(` you can use  a flow api  like  this:
```
value = (
image.sample(**{
     'region': bbox,
     'geometries': True
    })
    .first()
    .getInfo()
)
```

File: hydroengine_service/main.py [hunk 4]
```diff
@@ -1834,6 +1834,68 @@ def generate_image_info(im, params):
...
+        'region': bbox,
+        'geometries': True
+    }).first().getInfo()
+
+    if info_format == 'JSON':
+        if not value:
```
Comment (SiggyF) [inline hydroengine_service/main.py]: what  if `value == 0.0`

File: hydroengine_service/main.py (hunk 2 above)
Comment (ckrogers) [reply hydroengine_service/main.py]: 2 tests- 1 for values returned, one for null value

**Pull request merged**